// @draftbit/core/lib/module/components/Touchable.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import React from "react";
import { Pressable } from "react-native";
export default function Touchable(_ref) {
  let {
    children,
    disabled,
    onPress,
    activeOpacity = 0.8,
    disabledOpacity = 0.8,
    delayLongPress,
    hitSlop,
    style,
    ...props
  } = _ref;
  return /*#__PURE__*/React.createElement(Pressable, _extends({
    disabled: disabled,
    onPress: onPress,
    delayLongPress: delayLongPress ? delayLongPress : 500,
    hitSlop: hitSlop ? hitSlop : 8,
    style: _ref2 => {
      let {
        pressed
      } = _ref2;
      return [{
        opacity: pressed ? activeOpacity : disabled ? disabledOpacity : 1
      }, style];
    }
  }, props), children);
}
