// @draftbit/core/lib/commonjs/mappings/AudioPlayer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SEED_DATA = void 0;
var _types = require("@draftbit/types");
const SEED_DATA = {
  name: "Audio Player",
  tag: "AudioPlayer",
  description: "Given a source URL, plays sounds & audio!",
  category: _types.COMPONENT_TYPES.media,
  stylesPanelSections: [_types.StylesPanelSections.Typography, _types.StylesPanelSections.Background, _types.StylesPanelSections.Borders, _types.StylesPanelSections.Size, _types.StylesPanelSections.MarginsAndPaddings, _types.StylesPanelSections.Position],
  layout: {
    backgroundColor: "#eee",
    paddingLeft: 16,
    paddingRight: 16,
    paddingTop: 8,
    paddingBottom: 8,
    borderRadius: 24,
    flexDirection: "row",
    alignItems: "center"
  },
  props: {
    source: {
      group: _types.GROUPS.data,
      label: "Source",
      description: "The source URL for the audio file.",
      editable: true,
      required: true,
      defaultValue: "https://static.draftbit.com/audio/intro-to-draftbit-audio.mp3",
      formType: _types.FORM_TYPES.sourceUrl,
      propType: _types.PROP_TYPES.OBJECT
    },
    sliderColor: (0, _types.createColorProp)({
      label: "Thumb Color",
      defaultValue: "strong"
    }),
    completedTrackColor: (0, _types.createColorProp)({
      label: "Completed Track Color",
      defaultValue: "background"
    }),
    remainingTrackColor: (0, _types.createColorProp)({
      label: "Remaining Track Color",
      defaultValue: "light"
    }),
    trackThumbSize: (0, _types.createNumberProp)({
      label: "Thumb Size",
      defaultValue: 24
    }),
    playIconColor: (0, _types.createColorProp)({
      label: "Play Icon Color",
      defaultValue: "strong"
    })
  }
};
exports.SEED_DATA = SEED_DATA;
