// @draftbit/core/lib/commonjs/mappings/Button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SEED_DATA = void 0;
var _types = require("@draftbit/types");
const SEED_DATA_TRIGGERS = [_types.Triggers.OnPress, _types.Triggers.OnLongPress];
const SEED_DATA_PROPS = {
  onPress: (0, _types.createActionProp)(),
  onLongPress: (0, _types.createActionProp)(),
  icon: (0, _types.createIconProp)({
    defaultValue: null,
    required: false
  }),
  title: (0, _types.createTextProp)({
    label: "Label",
    description: "Button Label",
    defaultValue: "Get Started"
  }),
  disabled: (0, _types.createDisabledProp)(),
  loading: (0, _types.createLoadingProp)(),
  activeOpacity: (0, _types.createStaticNumberProp)({
    label: "Active Opacity",
    description: "Opacity of the button when active.",
    defaultValue: null,
    min: 0,
    max: 1,
    step: 0.01,
    precision: 2,
    required: false
  }),
  disabledOpacity: (0, _types.createStaticNumberProp)({
    label: "Disabled Opacity",
    description: "Opacity of the button when disabled.",
    defaultValue: null,
    min: 0,
    max: 1,
    step: 0.01,
    precision: 2,
    required: false
  }),
  delayLongPress: (0, _types.createStaticNumberProp)({
    label: "Delay Long Press",
    description: "Duration (in milliseconds) from onPressIn before onLongPress is called.",
    required: false
  }),
  hitSlop: (0, _types.createStaticNumberProp)({
    label: "Hit Slop",
    description: "Sets additional distance outside of element in which a press can be detected",
    required: false
  })
};
const SEED_DATA = [{
  name: "Button Outline",
  tag: "ButtonOutline",
  category: _types.COMPONENT_TYPES.deprecated,
  stylesPanelSections: [_types.StylesPanelSections.Typography, _types.StylesPanelSections.Background, _types.StylesPanelSections.Borders, _types.StylesPanelSections.Size, _types.StylesPanelSections.MarginsAndPaddings, _types.StylesPanelSections.Position, _types.StylesPanelSections.Effects],
  layout: {
    borderRadius: 8,
    fontFamily: "system-700",
    backgroundColor: "transparent",
    borderWidth: 1,
    textAlign: "center"
  },
  triggers: SEED_DATA_TRIGGERS,
  props: SEED_DATA_PROPS
}, {
  name: "Button Solid",
  tag: "ButtonSolid",
  category: _types.COMPONENT_TYPES.deprecated,
  stylesPanelSections: [_types.StylesPanelSections.Typography, _types.StylesPanelSections.Background, _types.StylesPanelSections.Borders, _types.StylesPanelSections.Size, _types.StylesPanelSections.MarginsAndPaddings, _types.StylesPanelSections.Position, _types.StylesPanelSections.Effects],
  layout: {
    borderRadius: 8,
    fontFamily: "system-700",
    backgroundColor: "primary",
    textAlign: "center"
  },
  triggers: SEED_DATA_TRIGGERS,
  props: SEED_DATA_PROPS
}, {
  name: "Button",
  tag: "Button",
  category: _types.COMPONENT_TYPES.button,
  stylesPanelSections: [_types.StylesPanelSections.Typography, _types.StylesPanelSections.Background, _types.StylesPanelSections.Borders, _types.StylesPanelSections.Size, _types.StylesPanelSections.MarginsAndPaddings, _types.StylesPanelSections.Position, _types.StylesPanelSections.Effects],
  layout: {
    borderRadius: 8,
    fontFamily: "system-700",
    backgroundColor: "primary",
    textAlign: "center"
  },
  triggers: SEED_DATA_TRIGGERS,
  props: SEED_DATA_PROPS
}];
exports.SEED_DATA = SEED_DATA;
