// @draftbit/core/lib/commonjs/mappings/FlatList.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SEED_DATA = void 0;
var _types = require("@draftbit/types");
const SEED_DATA = {
  name: "List",
  tag: "FlatList",
  description: "A basic List component",
  category: _types.COMPONENT_TYPES.data,
  stylesPanelSections: _types.CONTAINER_COMPONENT_STYLES_SECTIONS,
  layout: {
    flex: 1
  },
  triggers: [_types.Triggers.OnRefresh, _types.Triggers.OnEndReached],
  props: {
    onRefresh: (0, _types.createActionProp)(),
    onEndReached: (0, _types.createActionProp)(),
    horizontal: (0, _types.createStaticBoolProp)({
      label: "Horizontal",
      description: "Render list horizontally"
    }),
    inverted: (0, _types.createStaticBoolProp)({
      label: "Inverted",
      description: "If true, reverses the direction."
    }),
    numColumns: (0, _types.createNumColumnsType)({
      editable: true
    }),
    initialNumToRender: (0, _types.createStaticNumberProp)({
      label: "Initial Num To Render",
      description: "How many items to render in the initial batch",
      defaultValue: null
    }),
    onEndReachedThreshold: (0, _types.createStaticNumberProp)({
      label: "End Reached Threshold",
      description: "How far from the end (in units of visible length of the list) the bottom edge of the list must be from the end of the content to trigger the onEndReached callback. Thus a value of 0.5 will trigger onEndReached when the end of the content is within half the visible length of the list.",
      defaultValue: 0.5
    }),
    refreshColor: (0, _types.createColorProp)({
      label: "Refreshing Color",
      description: "Color of the refresh indicator"
    }),
    showsHorizontalScrollIndicator: (0, _types.createStaticBoolProp)({
      label: "Show Horizontal Scroll Indicator",
      description: "When true, shows a horizontal scroll indicator. The default value is true.",
      defaultValue: true
    }),
    showsVerticalScrollIndicator: (0, _types.createStaticBoolProp)({
      label: "Show Vertical Scroll Indicator",
      description: "When true, shows a vertical scroll indicator. The default value is true.",
      defaultValue: true
    })
  }
};
exports.SEED_DATA = SEED_DATA;
