// @draftbit/core/lib/commonjs/mappings/Image.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SEED_DATA = void 0;
var _types = require("@draftbit/types");
const SEED_DATA = {
  name: "Image",
  tag: "Image",
  description: "A basic Image Component",
  category: _types.COMPONENT_TYPES.media,
  stylesPanelSections: [_types.StylesPanelSections.Size, _types.StylesPanelSections.Margins, _types.StylesPanelSections.Position, _types.StylesPanelSections.Effects, _types.StylesPanelSections.Borders],
  layout: {
    width: 100,
    height: 100
  },
  props: {
    source: (0, _types.createImageProp)(),
    resizeMode: (0, _types.createResizeModeProp)(),
    blurRadius: {
      group: _types.GROUPS.basic,
      name: "blurRadius",
      label: "Blur Radius",
      defaultValue: null,
      description: "The blur radius of the blur filter added to the image.",
      editable: true,
      required: false,
      formType: _types.FORM_TYPES.number,
      propType: _types.PROP_TYPES.NUMBER,
      step: 5,
      precision: 1
    }
  }
};
exports.SEED_DATA = SEED_DATA;
