// @draftbit/core/lib/commonjs/mappings/Touchable.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SEED_DATA = void 0;
var _types = require("@draftbit/types");
const SEED_DATA_PROPS = {
  stylesPanelSections: [_types.StylesPanelSections.Size, _types.StylesPanelSections.Margins, _types.StylesPanelSections.Borders],
  layout: {},
  triggers: [_types.Triggers.OnPress, _types.Triggers.OnLongPress],
  props: {
    onPress: (0, _types.createActionProp)(),
    onLongPress: (0, _types.createActionProp)(),
    disabled: (0, _types.createDisabledProp)(),
    activeOpacity: (0, _types.createStaticNumberProp)({
      label: "Active Opacity",
      description: "The opacity when the button is pressed.",
      defaultValue: null,
      min: 0,
      max: 1,
      step: 0.01,
      precision: 2,
      required: false
    }),
    disabledOpacity: (0, _types.createStaticNumberProp)({
      label: "Disabled Opacity",
      description: "The opacity when the button is disabled.",
      defaultValue: null,
      min: 0,
      max: 1,
      step: 0.01,
      precision: 2,
      required: false
    }),
    delayLongPress: (0, _types.createStaticNumberProp)({
      label: "Delay Long Press",
      description: "Duration (in milliseconds) from onPressIn before onLongPress is called.",
      required: false
    }),
    hitSlop: (0, _types.createStaticNumberProp)({
      label: "Hit Slop",
      description: "Sets additional distance outside of element in which a press can be detected.",
      required: false
    }),
    android_disableSound: (0, _types.createStaticBoolProp)({
      label: "Disable Sound",
      description: "Disable the Android sound effect.",
      defaultValue: null,
      group: _types.GROUPS.android
    })
  }
};
const SEED_DATA = [{
  name: "Touchable",
  tag: "Touchable",
  description: "An interactive view with no styles",
  category: _types.COMPONENT_TYPES.button,
  ...SEED_DATA_PROPS
}, {
  name: "Pressable",
  tag: "Pressable",
  description: "An interactive view with no styles",
  category: _types.COMPONENT_TYPES.button,
  ...SEED_DATA_PROPS
}];
exports.SEED_DATA = SEED_DATA;
