// @draftbit/core/lib/commonjs/components/Pressable.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Pressable;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function Pressable(_ref) {
  let {
    children,
    disabled,
    onPress,
    activeOpacity = 0.8,
    disabledOpacity = 0.8,
    delayLongPress,
    hitSlop,
    style,
    ...props
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_reactNative.Pressable, _extends({
    onPress: onPress,
    disabled: disabled,
    delayLongPress: delayLongPress ? delayLongPress : 500,
    hitSlop: hitSlop ? hitSlop : 8,
    style: _ref2 => {
      let {
        pressed
      } = _ref2;
      return [{
        opacity: pressed ? activeOpacity : disabled ? disabledOpacity : 1
      }, style];
    }
  }, props), children);
}
