// @draftbit/core/lib/module/components/Swiper/Swiper.js
import React from "react";
import { View } from "react-native";
import SwiperComponent from "react-native-web-swiper";
const Swiper = _ref => {
  let {
    vertical = false,
    loop = false,
    timeout = 0,
    from = 0,
    prevTitle = "",
    nextTitle = "",
    prevTitleColor,
    nextTitleColor,
    dotsTouchable = true,
    dotColor,
    dotActiveColor,
    data,
    keyExtractor,
    renderItem,
    children,
    onIndexChanged,
    style
  } = _ref;
  return /*#__PURE__*/React.createElement(View, {
    style: style
  }, /*#__PURE__*/React.createElement(SwiperComponent, {
    from: from,
    loop: loop,
    timeout: timeout,
    vertical: vertical,
    onIndexChanged: onIndexChanged,
    controlsProps: {
      prevTitle,
      nextTitle,
      prevTitleStyle: {
        color: prevTitleColor
      },
      nextTitleStyle: {
        color: nextTitleColor
      },
      dotsTouchable,
      ...(dotColor ? {
        dotProps: {
          badgeStyle: {
            backgroundColor: dotColor
          }
        }
      } : {}),
      ...(dotActiveColor ? {
        dotActiveStyle: {
          backgroundColor: dotActiveColor
        }
      } : {})
    }
  }, data && renderItem ? data.map((item, index) => {
    const component = renderItem({
      item,
      index
    });
    if (!component) {
      return null;
    }
    const key = keyExtractor ? keyExtractor(item, index) : index;
    return /*#__PURE__*/React.cloneElement(component, {
      key
    });
  }) : children));
};
export default Swiper;
